lib/perl/Genome/InstrumentData/AlignmentResult.pm

```diff
@@ -1628,10 +1643,144 @@ sub verify_alignment_data {
     return 1;
 }
 
+# This will generally only return something until the first merge has deleted the original per-lane bam file
 sub alignment_bam_file_paths {
+    return glob(File::Spec->join(shift->output_dir, "*.bam"));
+}
+
+# This method will recreate the per-lane bam file and return the path.
+# This must be provided an allocation into which the bam will go (so that it can be cleaned up in the parent process once it is done).
+# The calling process is responsible for cleaning up the allocation after we are done with it.
+sub recreated_alignment_bam_file_paths {
+    my $self = shift;
+    my %p = Params::Validate::validate(@_, {disk_allocation => { isa => 'Genome::Disk::Allocation'}});
+
+    my @bams = $self->alignment_bam_file_paths;
+    if (@bams) {
+        return @bams;
+    }
+
+    my $recreated_bam = File::Spec->join($p{disk_allocation}->absolute_path, 'all_sequences.bam');
+    my $merged_bam    = $self->get_merged_bam_to_revivify_per_lane_bam;
+
+    unless ($merged_bam and -s $merged_bam) {
+        die $self->error_message('Failed to get valid merged bam to recreate per lane bam.');
+    }
+
+    my $cmd = Genome::InstrumentData::AlignmentResult::Command::RecreatePerLaneBam->create(
+        merged_bam          => $merged_bam,
+        per_lane_bam        => $recreated_bam,
+        instrument_data_id  => $self->instrument_data_id,
+        samtools_version    => $self->samtools_version,
+        picard_version      => $self->picard_version,
+        bam_header          => $self->bam_header_path,
+        comparison_flagstat => $self->flagstat_path,
+    );
+
+    unless ($cmd->execute) {
+        die $self->error_message('Failed to execute RecreatePerLaneBam');
+    }
+
+    if (-s $recreated_bam) {
+        return ($recreated_bam);
+    }
+    else {
+        die $self->error_message("After running RecreatePerLaneBam, no per-lane bam (%s) exists still!", $recreated_bam);
+    }
+}
+
+
+sub bam_header_path {
+    return File::Spec->join(shift->output_dir, 'all_sequences.bam.header');
+}
+
+
+sub flagstat_path {
+    return File::Spec->join(shift->output_dir, 'all_sequences.bam.flagstat');
+}
+
+
+sub get_merged_bam_to_revivify_per_lane_bam {
+    my $self = shift;
+    my $merged_result = $self->get_smallest_merged_alignment_result($self->get_unarchived_merged_alignment_results);
+
+    unless ($merged_result) {
+        $merged_result = $self->get_smallest_merged_alignment_result($self->get_merged_alignment_results);
```

Comment:
Is get_merged_alignment_results guaranteed to return at least one thing or die?  If not, we'll need to check here that merged_result actually exists before calling _auto_unarchive.